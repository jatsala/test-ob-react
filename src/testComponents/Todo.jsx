import React, { useState } from 'react';

const Todo = ({ todo }) => {

    const { id, text, completed } = todo;
    const [checked, setChecked] = useState(completed);

    return (
        <div data-testid={`todo-test-${id}`}>
            <label htmlFor="">
                <input type="checkbox" checked={checked} id='checkbox' onChange={() => setChecked(!checked)} />
                {text}
            </label>
        </div>
    )
}

export default Todo
