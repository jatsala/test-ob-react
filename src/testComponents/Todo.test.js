import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from './Todo';

afterEach(() => cleanup());

describe('Todo component is working correctly', () => {
    const todo = { id: 5, text: 'Hacer la compra', completed: false };
    render(<Todo todo={todo} />);
    const todoelement = screen.getByTestId(`todo-test-${todo.id}`);
    // eslint-disable-next-line testing-library/no-node-access
    // const checkbox = todoelement.querySelector('#checkbox');
    it('should render Todo component', () => {
        expect(todoelement).toBeInTheDocument();
    })
    it('should have right text', () => {
        expect(todoelement).toHaveTextContent(todo.text);
    })

    const todo2 = { id: 12, text: 'Hacer la compra', completed: true };
    render(<Todo todo={todo2} />);
    const todoelement2 = screen.getByTestId(`todo-test-${todo2.id}`);
    // eslint-disable-next-line testing-library/no-node-access
    const checkbox = todoelement2.querySelector('#checkbox');
    it('should be checked', () => {
        expect(checkbox).toBeChecked();
    })

    it('matches snapshot', () => {
        const snapTodo = { id: 1, text: 'Pasar los test', completed: false };
        const tree = renderer.create(<Todo todo={snapTodo} />);
        expect(tree).toMatchSnapshot();
    })
});



