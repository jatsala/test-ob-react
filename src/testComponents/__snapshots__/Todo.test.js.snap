// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component is working correctly matches snapshot 1`] = `
<div
  data-testid="todo-test-1"
>
  <label
    htmlFor=""
  >
    <input
      checked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    Pasar los test
  </label>
</div>
`;
